### PR TITLE
FIX No longer train batch-norm layer during pretraining

### PR DESCRIPTION
**Pull request by lucas-diedrich**
Fix the issue that the gradient of the batch norm layer was still incorrectly updated during pretraining. On some rare occasions (i.e. some random initializations), this led to the issue that the covariate-informed dimension did not capture the covariate-associated gene expression anymore. 

Updated
- Freeze batch norm layer during pretraining
- Updated unit tests. Unit tests now capture this unwanted behavior

### CODE REVIEW

**Comment by lucas-diedrich** [comment]
Failed test due to failed HTTP request for `test_data` (non-essential)

**Comment by lucas-diedrich** [comment]
Error in `test_data` - Remote dataset cannot be downloaded (non-essential) HTTP Error 403: Forbidden
